src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/QuiverStuff.kt

```diff
@@ -0,0 +1,355 @@
+
+            if (chestName == "Quiver") {
+                updateArrows(itemStacks.subList(0, 45))
+            } else if (chestName == "Forge Arrows") {
+                itemStacks.find { ItemUtil.getSkyBlockItemID(it) == "ARROW_SWAPPER" }?.let {
+                    val line = ItemUtil.getItemLore(it).find { lore -> lore.startsWith("§aSelected: ") }
+                    selectedType = arrowMap[line?.substringAfter("§aSelected: ")] ?: "UNKNOWN"
+                }
+            }
+        }
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST, receiveCanceled = true)
+    fun onChatReceived(event: ClientChatReceivedEvent) {
+        if (event.type == 2.toByte() || !Utils.inSkyblock) return
+        val formatted = event.message.formattedText
+
+        if (clearQuiverRegex.matches(formatted)) {
+            quiver.clear()
+        }
+
+        if (emptyQuiverRegex.matches(formatted)) {
+            if (quiver.isNotEmpty()) {
+                quiver.clear()
+            }
+        }
+
+        littleArrowsLeft.find(formatted)?.let { result ->
+            val amount = result.groups["amount"]?.value?.toIntOrNull() ?: return@let
+
+            arrowCount = amount
+        }
+
+        arrowRefillRegex.find(formatted)?.let {
+            val amount = it.groups["amount"]?.value?.toIntOrNull() ?: return@let
+
+            setArrows(getArrows("ARROW") + amount, "ARROW")
+        }
+
+        jaxForgedRegex.find(formatted)?.let {
+            val forged = arrowMap[it.groups["type"]?.value] ?: "UNKNOWN"
+            val amount = it.groups["amount"]?.value?.toIntOrNull() ?: return@let
+
+            setArrows(getArrows(forged) + amount, forged)
+        }
+
+        selectedArrowRegex.find(formatted)?.let { result ->
+            val selected = result.groups["selected"]?.value
+            selectedType = arrowMap.entries.find { it.key.substring(2) == selected?.substring(2) }?.value ?: "UNKNOWN"
+
+            if (Skytils.config.restockArrowsWarning != 0 && arrowCount < Skytils.config.restockArrowsWarning) {
+                sendWarning()
+            }
+        }
+
+        if (resetPreferenceRegex.matches(formatted)) {
+            selectedType = "NONE"
+        }
+    }
+
+    @SubscribeEvent(priority = EventPriority.LOWEST)
+    fun onReceivePacket(event: MainReceivePacketEvent<*, *>) {
+        if (!Utils.inSkyblock) return
+        if (event.packet is S29PacketSoundEffect) {
+            val packet = event.packet
+            val sound = packet.soundName
+            val volume = packet.volume
+
+            if (sound == "random.bow" && volume == 1f && arrowCount > 0 && Skytils.mc.thePlayer.heldItem.item == Items.bow) {
+                val armor = (3 downTo 0).mapNotNull { Skytils.mc.thePlayer.getCurrentArmor(it) }
+                val extraAttr = ItemUtil.getExtraAttributes(Skytils.mc.thePlayer.heldItem)
+                if (!armor.any { it.displayName.contains("Skeleton Master") } && extraAttr != null) {
```

Comment:
Only Skeleton Master Chestplate makes it so you dont consume arrows, other pieces dont